src/index.ts

```diff
@@ -373,22 +378,19 @@ task("starknet-deploy", "Deploys Starknet contracts which have been compiled.")
 
 
 async function findPath(traversable: string, name: string) {
-    let foundPath: string;
-    let paths : string[] = [];
-    if(fs.lstatSync(traversable).isDirectory()){
-        paths = await glob(path.join(traversable,"**","*"));
-    }
-    else{
-        paths.push(traversable);
-    }
-    paths.filter(file => {
+    let files = await traverseFiles(traversable);
+    files = files.filter(file => {
         return path.basename(file) === name;
-    }).forEach(file => {
-        foundPath = file;
-        return 0;
     });
-    
-    return foundPath;
+    if(files.length == 0){
+        return null;
+    }
+    else if(files.length == 1){
+        return files[0];
+    }
+    else {
+        throw new HardhatPluginError(PLUGIN_NAME, `More than one file was found because the path provided is ambiguous, please specify a relative path`);
```

Comment:
This is a rather long line. Handle it as in other such places of this project. Also no need for the template literal (the backtick string definition) if nothing is interpolated with `${}`)

Reply:
Also, does providing relative path work? We should have it tested in starknet-hardhat-example